tinyagentos/chat/message_store.py

```
@@ -199,6 +199,38 @@ async def delete_message(self, message_id: str) -> bool:
+            params = (now, channel_id, after_timestamp, thread_id)
+        else:
+            thread_clause = "AND thread_id IS NULL"
+            params = (now, channel_id, after_timestamp)
+        # NOTE: strict > means a message sharing the edited message's exact
+        # time.time() survives — at most one stale adjacent message can
+        # slip through a timestamp tie.
+        cursor = await self._db.execute(
+            f"""UPDATE chat_messages SET deleted_at = ?
+               WHERE channel_id = ? AND created_at > ? AND deleted_at IS NULL
+               {thread_clause}
+               RETURNING id""",
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)